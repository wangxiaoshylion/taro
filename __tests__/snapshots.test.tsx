import React from 'react';
import renderer from 'react-test-renderer';
import Index from '@/pages/index/index';

it('toMatchSnapshot', () => {
  const tree = renderer.create(<Index />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('toMatchInlineSnapshot', () => {
  const tree = renderer.create(<Index title="内联快照" />).toJSON();
  expect(tree).toMatchInlineSnapshot(`
    <taro-view-core>
      <taro-text-core>
        内联快照
      </taro-text-core>
    </taro-view-core>
  `);
});

it.only('will fail every time', () => {
  // 对象
  const user = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
  };
  expect(user).toMatchInlineSnapshot(
    {
      createdAt: expect.any(Date),
      id: expect.any(Number), // 忽略掉这俩变化
    },
    `
    {
      "createdAt": Any<Date>,
      "id": Any<Number>,
      "name": "LeBron James",
    }
  `,
  );

  // 字符串
  const randomNumber = Math.round(Math.random() * 100);
  const stringWithRandomData = `<div id="${randomNumber}">Lorem ipsum</div>`;
  const stringWithConstantData = stringWithRandomData.replace(/id="\d+"/, 'id="idNumber"');
  console.log(stringWithConstantData);
  expect(stringWithConstantData).toMatchSnapshot();
});
