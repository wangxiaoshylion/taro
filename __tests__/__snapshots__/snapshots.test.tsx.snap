// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toMatchSnapshot 1`] = `
<taro-view-core>
  <taro-text-core>
    Hello world!
  </taro-text-core>
</taro-view-core>
`;

exports[`will fail every time 2`] = `"<div id="idNumber">Lorem ipsum</div>"`;
